src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,186 @@
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordExists_NotOk() throws RegistrationException {
+        User user = new User();
+        user.setLogin("existingUser");
+        user.setPassword("password123");
+        user.setAge(21);
+        storageDao.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() throws RegistrationException {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void registerLoginExists_NotOk() throws RegistrationException {
+        User existingUser = new User();
+        existingUser.setLogin("existingUser");
+        existingUser.setPassword("password123");
+        existingUser.setAge(20);
+        storageDao.add(existingUser);
+
+        User newUserWithExistingLogin = new User();
+        newUserWithExistingLogin.setLogin("existingUser");
+        newUserWithExistingLogin.setPassword("newPassword123");
+        newUserWithExistingLogin.setAge(25);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUserWithExistingLogin);
+        });
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerPassword_NotOk() throws RegistrationException {
+        String[] invalidPasswords = {"1", "ab", "son", "user", "hell0"};
+        for (String password : invalidPasswords) {
+            user.setLogin("validLog");
+            user.setPassword(password);
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Password length must be at least 6 characters.", exception.getMessage());
+        }
```

Comment:
Similar to the login length test, this test case for password length should also include the edge cases. You should add a test for a password with exactly 6 characters to make sure that the minimum valid length is accepted.